devDocs/developerGuide.t2t

```diff
@@ -947,3 +947,103 @@ There are some special functions:
 - snap(): Takes a snapshot of the current state of NVDA and saves it in the [snapshot variables #PythonConsoleSnapshotVariables].
 - rmSnap(): Removes all snapshot variables.
 -
+
++ Extension Points +[ExtensionPoints]
+NVDA's ``extensionPoints`` module, allows code in different parts of NVDA, or in add-ons, to perform tasks such as:
+- Be notified when an action occurs or a state is changed.
+- Receive, as part of being notified, variables related to the action or changed state.
+- Cancel or alter an action NVDA was going to take, based upon certain conditions.
+- Modify data that NVDA is using (such as changing speech sequences or braille, before they are spoken or brailled).
+- Delay something NVDA is doing, while intervening operations are performed.
+-
+
+There are five kinds of extension point:
+
+|| Type | Purpose |
+| ``Action`` | Allows some code to find out what other code is doing. For example, an add-on can be notified before or after a config profile changes. |
+| ``Filter`` | Edits data. A filter registered in the speech module, might allow changing speech strings before they are spoken. |
+| ``Decider`` | Runs each registered handler until one of them returns ``False``. If one does, it can be used to prevent the invoking code from running. |
+| ``AccumulatingDecider`` | Like ``Decider``, but always runs all of its registered handlers, and only decides if one of them failed at the end. The default decision is ``True`` by default, but can be set to ``False``. |
+| ``Chain`` | Allows registering handlers that return iterables (mainly generators). Calling ``iter`` on the ``Chain`` returns a generator that iterates over all the handlers. |
+
+The sections below provide the list of currently defined extension points in NVDA, along with brief descriptions for them.
+Please see code documentation in the associated files, or the code itself, for further explanation.
+The section titles below represent the package or module in which the listed extension points are defined.
+
+For examples of how to define and use new extension points, please see the code documentation of the ``extensionPoints`` module.
+
+++ braille ++[brailleExtPts]
+|| Type | Extension Point | Description |
+| ``Filter`` | ``filter_displaySize`` | Allows components or add-ons to change the display size used for braille output. |
+| Action | displaySizeChanged | Notifies of display size changes. |
+| Action | pre_writeCells | Notifies when cells are about to be written to a braille display |
+| Action | displayChanged | Notifies of braille display changes. |
+| Decider | decide_enabled | Allows deciding whether the braille handler should be forcefully disabled. |
```

Comment:
there inconsistent code formatting here. I think it might be worth committing to code formatting for all NVDA python symbols